### PR TITLE
Wording the next and last run DAG columns better

### LINKED ISSUE #31365: The `Next Run` column name and tooltip is misleading
### Description

> Expected date/time of the next DAG Run, or for dataset triggered DAGs, how many datasets have been updated since the last DAG Run

"Expected date/time of the next DAG Run" to me sounds like Run After.

Should the tooltip indicate something along the lines of "start interval of the next dagrun" or maybe the header Next Run is outdated? Something like "Next Data Interval"?

On the same vein, "Last Run" is equally as confusing. The header can be "Last Data Interval" in additional to a tool tip that describe it is the data interval start of the last dagrun.

### Use case/motivation

Users confused "Next Run" as when the next dagrun will be queued and ran and does not interpret it as the next dagrun's data interval start.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by amoghrajesh**
Users tend to get confused by the words `Next Run` and `Last Run` used in our DAG UI. These words and tooltips can be worded better to avoid getting misleading as users. Related discussions:
https://github.com/apache/airflow/pull/17732
https://github.com/apache/airflow/issues/21493

closes: #31365 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
@bbovenzi @potiuk need a review on this issue. I also see that the CI is failing and I cannot figure out what the issue. Any help here will be appreciated.

File: airflow/www/templates/airflow/dags.html [hunk 1]
```diff
@@ -215,13 +215,13 @@ <h2>{{ page_title }}</h2>
...
           </th>
           <th>Schedule</th>
-          <th style="width:180px;">{{ sortable_column("Last Run", "last_dagrun") }}
+          <th style="width:180px;">{{ sortable_column("Last Data Interval", "last_dagrun_data_interval") }}
             <span class="material-icons text-muted js-tooltip" aria-hidden="true"
-                  title="Date/Time of the latest Dag Run.">info</span>
+                  title="Starting Date/Time interval of the latest Dag Run.">info</span>
           </th>
-          <th style="width:180px;">{{ sortable_column("Next Run", "next_dagrun") }}
+          <th style="width:180px;">{{ sortable_column("Next Data Interval", "next_dagrun_data_interval") }}
```
Comment (bbovenzi) [inline airflow/www/templates/airflow/dags.html]: I don't think we want to change the values, just the label, right?

Comment (amoghrajesh) [reply airflow/www/templates/airflow/dags.html]: Do you mean that `Last Run` is the label and the `last_dagrun` is the value? 
I went ahead and made this modification as it seemed more appropriate to change both rather than just one of them as they are related. Let me keep the changes to label only

**Comment by bbovenzi** [comment]
Code looks good.
@wolfier what do you think?

**Comment by amoghrajesh** [comment]
@bbovenzi @wolfier this PR is ready for review. Can i get a round of review on this?

File: airflow/www/templates/airflow/dags.html [hunk 2]
```diff
@@ -215,13 +215,13 @@ <h2>{{ page_title }}</h2>
                   title="Status of all previous DAG runs.">info</span>
           </th>
           <th>Schedule</th>
-          <th style="width:180px;">{{ sortable_column("Last Run", "last_dagrun") }}
+          <th style="width:180px;">{{ sortable_column("Last Data Interval", "last_dagrun") }}
```
Comment (jedcunningham) [inline airflow/www/templates/airflow/dags.html]: If we want to be pedantic here, isn't this the logical date?

File: airflow/www/templates/airflow/dags.html [hunk 3]
```diff
@@ -215,13 +215,13 @@ <h2>{{ page_title }}</h2>
...
             <span class="material-icons text-muted js-tooltip" aria-hidden="true"
-                  title="Date/Time of the latest Dag Run.">info</span>
+                  title="Starting Date/Time interval of the latest Dag Run.">info</span>
```
Comment (jedcunningham) [inline airflow/www/templates/airflow/dags.html]: I find this confusing. I'd prefer we don't change the title (leave it "Last Run") and change this to "Logical Date of the latest Dag Run.".

Comment (amoghrajesh) [reply airflow/www/templates/airflow/dags.html]: Yeah makes sense to me because it is theoretically the logical date :)

File: airflow/www/templates/airflow/dags.html (hunk 2 above)
Comment (amoghrajesh) [reply airflow/www/templates/airflow/dags.html]: Yes it is. Incorporating the next comment in a commit

**Comment by amoghrajesh** [comment]
@jedcunningham @bbovenzi can I get another round of review on this PR?

File: tests/utils/test_db.py [hunk 4]
```diff
@@ -230,7 +230,7 @@ def test_resetdb(
         session_mock = MagicMock()
         resetdb(session_mock, skip_init=skip_init)
         mock_drop_airflow.assert_called_once_with(mock_connect.return_value)
-        mock_drop_moved.assert_called_once_with(session_mock)
+        mock_drop_moved.assert_called_once_with(mock_connect.return_value)
```
Comment (jedcunningham) [inline tests/utils/test_db.py]: This seems unrelated?

Comment (amoghrajesh) [reply tests/utils/test_db.py]: Oh, looks like the rebase added it. I will undo this

**Review by jedcunningham — APPROVED** [review]

**Review by bbovenzi — APPROVED** [review]
